### ISSUE TITLE
Devscripts: Updated fixes.patch for 2.21.2

### ISSUE THREAD

**Issue by selmf** [ItachiSan/linux_packaging#8]
The following code is an updated version of the fixes.patch patchfile used for the AUR package.

Main changes:

- Updated for use with devscripts 2.21.2
- Hardcoded docbook versions in the paths replaced by dynamic shell code as suggested on [the Arch Wiki](https://wiki.archlinux.org/title/DocBook)

Please note: This is a quick and dirty update and I have **not** updated the "diff --" and index lines. You may want to
recreate the patch from scratch to correct this.

``` diff
diff --git a/po4a/Makefile b/po4a/Makefile
index f19b24df..6a43f037 100644
--- a/po4a/Makefile
+++ b/po4a/Makefile
@@ -34,7 +34,7 @@
 	pod2man --utf8 --center=" " --release="$(DESC_$(dir $@))" $< > $@
 %.1:: %.dbk translate
 	xsltproc --nonet \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	 /usr/share/xml/docbook/xsl-stylesheets-"$(shell pacman -Q docbook-xsl | cut -d ' ' -f 2 | cut -d '-' -f 1)"/manpages/docbook.xsl $<
 	# /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/other.xsl
 	# (which is imported by the above stylesheet) insists in writing the output
 	# to where it wants to.  we can only move the file ourselves.

diff --git a/scripts/Makefile b/scripts/Makefile
index ece5455a..78bdb942 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -64,7 +64,7 @@ endif
 	pod2man --utf8 --center=" " --release="Debian Utilities" $< > $@
 %.1: %.dbk
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-"$(shell pacman -Q docbook-xsl | cut -d ' ' -f 2 | cut -d '-' -f 1)"/manpages/docbook.xsl $<
 
 # There is a slight chance this gets called twice, once here from here and once
 # from ../po4a/Makefile. Treat files with care.
@@ -93,7 +93,7 @@ test:
 	$(foreach python,$(shell py3versions -r ../debian/control),$(python) setup.py test$(\n))
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	python3 setup.py install --root="$(DESTDIR)" --no-compile
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
```

**Comment by ItachiSan** [comment]
Hi! I was quite busy recently, thus I had no time for working on side project.

Thanks for the input! The shell interpolation and the layout install improvements are definitely well welcomed 😄 

I will try the changes as soon as I have some time in the next days

**Comment by selmf** [comment]
Happy you like it :) Don't worry about integrating it fast, I only use the Devscripts once or twice a year when I have to update that one Debian package I am responsible for because the project I am involved in does not have a .deb downstream. When packaging for Debian from Arch you always run into some troubles and the Devscripts package saves me more than it generates.

**Comment by ItachiSan** [comment]
A new package is up, including all the changes (these fixes and the `optdepends` mentioned in #9 ), lemme know if these work for you 😄

**Comment by selmf** [comment]
I gave it a test and it seems to work fine. Thank you for including it :)
